src/libAtomVM/interop.c

```diff
@@ -176,19 +176,33 @@ term interop_proplist_get_value_default(term list, term key, term default_value)
     return default_value;
 }
 
+static InteropFunctionResult size_fold_fun(term t, void *accum)
+{
+    size_t *size = (size_t *) accum;
+    if (term_is_integer(t)) {
+        *size += 1;
+    } else if (term_is_binary(t)) {
+        *size += term_binary_size(t);
+    }
+    return InteropOk;
+}
+
 InteropFunctionResult interop_iolist_size(term t, size_t *size)
+{
+    *size = 0;
+    return interop_iolist_fold(t, size_fold_fun, (void *) size);
+}
+
+InteropFunctionResult interop_iolist_fold(term t, interop_iolist_fold_fun fold_fun, void *accum)
```

Comment:
The function seems to accept a binary, so it should probably be renamed `interop_iodata_fold`.

Reply:
I agree, but there would be two issues:
1. If I change the terminology of this suite of functions, it looks awkward and leads to the impression that other functions do not support raw binaries, as well;
2. If I change all the functions, this has a larger impact radius than I had planned with this PR

I guess I would prefer we merge this and do a separate PR with a renaming across the board.